rero_ils/modules/loans/api.py

```diff
@@ -290,6 +290,122 @@ def date_fields2str(self):
             if field in self:
                 self[field] = self[field].isoformat()
 
+    @classmethod
+    def requested_loans_to_validate(cls, library_pid):
+        """Get Requests to be validated."""
+        from ..item_types.api import ItemTypesSearch
+        from ..items.api import ItemsSearch
+
+        # Proxy record
+        patrons = {}
+        locations = {}
+        libraries = {}
+        items = {}
+        item_types = {}
+
+        def loans_pending():
+            """Get pending loans."""
+            return LoansSearch()\
+                .params(preserve_order=True)\
+                .filter('term', state=LoanState.PENDING)\
+                .filter('term', library_pid=library_pid)\
+                .sort({'_created': {"order": 'asc'}})\
+                .source(includes=[
+                    'pid', 'transaction_date', 'item_pid', 'patron_pid',
+                    'document_pid', 'library_pid', 'state',
+                    'transaction_location_pid', 'pickup_location_pid'
+                ])\
+                .scan()
+
+        def patron_by_pid(patron_pid):
+            """Get patron by pid."""
+            if patron_pid not in patrons:
+                patrons[patron_pid] = PatronsSearch()\
+                    .filter('term', pid=patron_pid)\
+                    .source(includes=[
+                        'pid', 'first_name', 'last_name', 'patron.barcode'])\
+                    .execute()[0].to_dict()
```

Comment:
It should not happen in the best world .... but this will cause Exception if patron_pid doesn't exists. By example if we delete patron/item/location and not the corresponding request.